sql-plugin/src/main/scala/com/nvidia/spark/rapids/GpuOrcScan.scala

```diff
@@ -641,12 +641,16 @@ class GpuOrcPartitionReader(
       }
       isFirstBatch = false
     }
-    batch.isDefined
+    val ret = batch.isDefined
+    // After finishing reading, we should clean up ctx just in case leaking orc readers
+    if (!ret) {
+      cleanUpOrc(ctx)
+    }
+    ret
```

Comment:
nit:
```suggestion
   batch.map(_ => true).getOrElse {
      // After finishing reading, we should clean up ctx just in case leaking orc readers
       cleanUpOrc(ctx)
       false
   }
```

Reply:
can we add a fix to cleanUpOrc to do a `safeClose` to avoid leaks too?
```scala
  import com.nvidia.spark.rapids.RapidsPluginImplicits._
  ...
  def cleanUpOrc(ctx: OrcPartitionReaderContext) = {
    if (ctx != null) {
      Seq(ctx.orcReader, ctx.dataReader).safeClose()
    }
  }
```

Reply:
Thx for the suggestion. Done